agent/virus_total_agent.py

```diff
@@ -87,13 +87,31 @@ def _process_response(self, response: dict[str, Any], target: str | None) -> Non
         scans = virustotal.get_scans(response)
         try:
             if scans is not None:
-                technical_detail = process_scans.get_technical_details(scans, target)
-                risk_rating = process_scans.get_risk_rating(scans)
-                self.report_vulnerability(
-                    entry=kb.KB.VIRUSTOTAL_SCAN,
-                    technical_detail=technical_detail,
-                    risk_rating=risk_rating,
-                )
+                (
+                    secure_scan_report,
+                    vulnerable_scan_report,
+                ) = process_scans.split_scans_by_result(scans)
+
+                if len(secure_scan_report) > 0:
+                    technical_detail = process_scans.get_technical_details(
+                        secure_scan_report, target
+                    )
+                    self.report_vulnerability(
+                        entry=kb.KB.SECURE_VIRUSTOTAL_SCAN,
+                        technical_detail=technical_detail,
+                        risk_rating=agent_report_vulnerability_mixin.RiskRating.SECURE,
+                    )
+
+                if len(vulnerable_scan_report) > 0:
+                    technical_detail = process_scans.get_technical_details(
+                        vulnerable_scan_report, target
+                    )
+                    self.report_vulnerability(
+                        entry=kb.KB.INSECURE_VIRUSTOTAL_SCAN,
```

Comment:
Would be better to add scan link in KB references rather than `www.virustotal.com`